cpp/include/cudf/detail/gather.cuh

```diff
@@ -450,16 +450,24 @@ struct column_gatherer_impl<struct_view> {
                                      rmm::cuda_stream_view stream,
                                      rmm::mr::device_memory_resource* mr)
   {
-    structs_column_view structs_column(column);
-    auto gather_map_size{std::distance(gather_map_begin, gather_map_end)};
+    auto const gather_map_size = std::distance(gather_map_begin, gather_map_end);
```

Comment:
Changes in this `gather.cuh` file is not part of this PR. Instead, it is upstream from another PR (https://github.com/rapidsai/cudf/pull/9218). Please don't review it here.